frontend/src/js/previous-queries/list/ShareProjectItemModal.tsx

```diff
@@ -0,0 +1,156 @@
+import styled from "@emotion/styled";
+import { StateT } from "app-types";
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+
+import type { SelectOptionT, UserGroupT } from "../../api/types";
+import PrimaryButton from "../../button/PrimaryButton";
+import { TransparentButton } from "../../button/TransparentButton";
+import Modal from "../../modal/Modal";
+import InputMultiSelect from "../../ui-components/InputMultiSelect/InputMultiSelect";
+
+import type { ProjectItemT } from "./ProjectItem";
+import {
+  useLoadQuery,
+  useLoadFormConfig,
+  useUpdateQuery,
+  useUpdateFormConfig,
+} from "./actions";
+import { isFormConfig } from "./helpers";
+
+const Buttons = styled("div")`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const SxPrimaryButton = styled(PrimaryButton)`
+  margin-left: 20px;
+`;
+
+const SxInputMultiSelect = styled(InputMultiSelect)`
+  display: block;
+  margin-bottom: 20px;
+`;
+
+const QueryName = styled("p")`
+  margin: -15px 0 20px;
+`;
+
+const getUserGroupsValue = (
+  userGroups: UserGroupT[],
+  projectItem?: ProjectItemT,
+) => {
+  return projectItem && projectItem.groups
+    ? userGroups
+        .filter((group) => projectItem.groups?.includes(group.id))
+        .map((group) => ({
+          label: group.label,
+          value: group.id,
+        }))
+    : [];
+};
+
+interface PropsT {
+  item: ProjectItemT;
+  onClose: () => void;
+}
+
+const ShareProjectItemModal = ({ item, onClose }: PropsT) => {
+  const { t } = useTranslation();
+  const userGroups = useSelector<StateT, UserGroupT[]>((state) =>
+    state.user.me ? state.user.me.groups : [],
+  );
+
+  const initialUserGroupsValue = getUserGroupsValue(userGroups, item);
+
+  const [userGroupsValue, setUserGroupsValue] = useState<SelectOptionT[]>(
+    initialUserGroupsValue,
+  );
+
+  const [loadedOnce, setLoadedOnce] = useState(false);
+
+  const { loadQuery } = useLoadQuery();
+  const { loadFormConfig } = useLoadFormConfig();
+  const { updateQuery } = useUpdateQuery();
+  const { updateFormConfig } = useUpdateFormConfig();
+
+  useEffect(
```

Comment:
hier mischt du js und ts function notation

Reply:
There's no such thing as js and ts function notation, there are named functions and arrow functions, they're both part of js and ts, and I've turned to use named functions in `useEffect` wherever I can to document what the function does, unless it's super clear